### PR TITLE
Admin: Customize Shuup admin panel menu

### PR DESCRIPTION
**Pull request by volodymyr-code**

### CODE REVIEW

**Comment by tulimaki** [comment]
- Consider making the 2nd level dragging are a little bit bigger
- Deprecate `MenuItem.subcategories` no need to group menu items based on menu category
- Maybe we could change the module from `shuup.admin.modules.admin_menu` to `shuup.admin.modules.menu`

File: shuup/admin/base.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 # LICENSE file in the root directory of this source tree.
 from __future__ import unicode_literals
 
+import abc
```
Comment (tulimaki) [inline shuup/admin/base.py]: Where is this used?

**Comment by tulimaki** [comment]
```
diff --git a/shuup/core/utils/menu.py b/shuup/core/utils/menu.py
index e75ca29fa..70e98b91d 100644
--- a/shuup/core/utils/menu.py
+++ b/shuup/core/utils/menu.py
@@ -28,6 +28,9 @@ class MainMenuUpdater(object):
         """
         for item in self.menu:
             for child in self.updates.get(item["identifier"], []):
-                if child not in item["children"]:
-                    item["children"].append(child)
+                if child not in item.get("children", []):
+                    if "children" not in item:
+                        item["children"] = [child]
+                    else:
+                        item["children"].append(child)
         return self.menu
```
Maybe something like this is required.

File: shuup/admin/base.py (hunk 1 above)
Comment (volodymyr-code) [reply shuup/admin/base.py]: fixed

File: shuup_tests/core/test_admin_menu.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from shuup.admin.menu import MAIN_MENU
+from shuup.testing.admin_main_menu_updater import SubscriptionMainMenuUpdater
+
+
+def test_menu_updater():
+    updater = SubscriptionMainMenuUpdater(MAIN_MENU)
```
Comment (tulimaki) [inline shuup_tests/core/test_admin_menu.py]: Use the "overrride_provides" instead. First test the "get menu" without this "menu updater" and then with and make sure those updates actually happened.

Comment (volodymyr-code) [reply shuup_tests/core/test_admin_menu.py]: fixed

**Comment by maximlion** [comment]
Having inactive item under an item makes the parent item unusuable. (Dropdown is active so user is not redirected anywhere) ( happens only when parent item has only one item)
Putting originally parent orders under child orders shows None in Orders dropdown. Parent items shouldn't ever be allowed to be child items. Add (parent) to the name of the initial parent items 
Screenshot from 2019-05-16 22-41-26

**Comment by volodymyr-code** [comment]
> Having inactive item under an item makes the parent item unusuable. (Dropdown is active so user is not redirected anywhere) ( happens only when parent item has only one item)
> Putting originally parent orders under child orders shows None in Orders dropdown. Parent items shouldn't ever be allowed to be child items. Add (parent) to the name of the initial parent items
> Screenshot from 2019-05-16 22-41-26

it is future, made for flexibility: we can realize any structure, for example we can move addons under settings or new plugin like "products multivendor" move under addons. I think will be better if we fix admin menu template and activate parent link for your case.

File: shuup/admin/static_src/base/js/admin-menu.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+const sortUpdate = (e) => {
+  const menuEntry = $(e.detail.item);
+  // update parent and entrye menu-entries sorter
+  connectSortable(menuEntry.parent());
+  connectSortable(menuEntry.find("> .menu-entries"));
+}
+
+const connectSortable  = (menuEntries) => {
```
Comment (chessbr) [inline shuup/admin/static_src/base/js/admin-menu.js]: double space here before `=`

**Comment by chessbr** [comment]
I removed Orders from the Order category to the root level and this is what is rendered:
image

The Orders menu should list the Carts as a child here.

**Pull request merged**